enh: Human-readable axis values/units on con-duct plots

asmacdo opened the pull request:
Fixes #291 

Looks good at first glance, I'd like try with a longer run before ready for review.

tested with a script to consume resources

<details><summary> duct --r-i 5 ./resource_consumer.py --demo</summary>

```
2025-09-09T12:54:16-0500 [INFO    ] con-duct: duct 0.15.0.post10+gf0c4796.d20250904 is executing './resource_consumer.py --demo'...
2025-09-09T12:54:16-0500 [INFO    ] con-duct: Log files will be written to .duct/logs/2025.09.09T12.54.16-679331_
Starting resource consumption test with 7 phases
Process PID: 679334
You can monitor this process with tools like htop, top, or your resource monitor

=== Phase: Baseline (Duration: 5s) ===
Running for 5 seconds...
CPU consumption completed

=== Phase: Low Memory (Duration: 10s) ===
Allocating 50MB RSS...
RSS allocated: 50MB
Mapping 100MB VSS...
VSS mapped: 100MB
Running for 10 seconds...
CPU consumption completed

=== Phase: Add CPU Load (Duration: 10s) ===
Starting CPU consumption: 2 cores at 50.0% for 10s...
Running for 10 seconds...
CPU consumption completed

=== Phase: High Memory + CPU (Duration: 15s) ===
Allocating 200MB RSS...
RSS allocated: 200MB
Mapping 300MB VSS...
VSS mapped: 300MB
Starting CPU consumption: 4 cores at 80.0% for 15s...
Running for 15 seconds...
CPU consumption completed

=== Phase: Memory Spike (Duration: 5s) ===
Allocating 500MB RSS...
RSS allocated: 500MB
Running for 5 seconds...
CPU consumption completed

=== Phase: Release Memory (Duration: 10s) ===
Starting CPU consumption: 1 cores at 30.0% for 10s...
Releasing 500MB RSS...
Running for 10 seconds...
CPU consumption completed

=== Phase: Cool Down (Duration: 5s) ===
Running for 5 seconds...
CPU consumption completed

=== Test Completed ===
Cleaning up resources...
Cleanup complete
2025-09-09T12:55:17-0500 [INFO    ] con-duct: Summary:
Exit Code: 0
Command: ./resource_consumer.py --demo
Log files location: .duct/logs/2025.09.09T12.54.16-679331_
Wall Clock Time: 60.800 sec
Memory Peak Usage (RSS): 1.4 GB
Memory Average Usage (RSS): 631.3 MB
Virtual Memory Peak Usage (VSZ): 4.7 GB
Virtual Memory Average Usage (VSZ): 2.0 GB
Memory Peak Percentage: 4.00%
Memory Average Percentage: 1.81%
CPU Peak Usage: 36.80%
Average CPU Usage: 10.48%

con-duct plot .duct/logs/2025.09.09T12.54.16-679331_info.json -o output

```

</details>

<img width="640" height="480" alt="output" src="https://github.com/user-attachments/assets/685865ba-765c-4692-888b-00194978854d" />

TODO:
- [x] add tests

yarikoptic commented:
For plots - across all values so guided by maximal value, no more than two units indeed and could even be 1 (with max two digits after decimal). For individual records - could be per record

asmacdo commented:
Since we need to pick a unit and use it for the whole plot, we can't use the existing SummaryFormatter. I've rewritten with a new custom formatter for Axes. As a bonus, for interactive environments this will change the units when you zoom in! 

I've also added a new config value  `min_ratio` which basically means "at least this many units should be present" ie, if you only have 2 minutes, continue to use seconds, or 2 GB use MB, but for 3 min/3GB use minutes and GB. 

Heres where we're at now:

--min-ratio 5 
<img width="640" height="480" alt="min-ratio-5" src="https://github.com/user-attachments/assets/079e90f1-a5d0-4e50-894e-cbc9d535c0c4" />

--min-ratio 3 (default)
<img width="640" height="480" alt="min-ratio-3" src="https://github.com/user-attachments/assets/0eb6b320-2907-4b27-aa4e-5ccbb21ef98b" />

--min-ratio 1
<img width="640" height="480" alt="min-ratio-1" src="https://github.com/user-attachments/assets/9efe7c19-9bca-4277-82cc-a6ae0f251a3f" />


--min-ratio -1 (use base units)
<img width="640" height="480" alt="min-ratio-base" src="https://github.com/user-attachments/assets/9fab6ed0-8de7-4f0b-b383-52c8526c2e40" />

asmacdo commented:
I could see a case that the user would want independent control, we could separate these values, ie `min-ratio-time`/`min-ratio-mem`

yarikoptic commented:
I think this is great for now!  likely no even need to separate ATM
